esx_service/cli/vmdkops_admin.py

```diff
@@ -1063,17 +1094,63 @@ def tenant_access_set(args):
     if not privileges:
         error_info = "No privileges exist for ({0}, {1})".format(args.name, args.datastore)
         return operation_fail(error_info)
-    
+
     privileges_dict = generate_privileges_dict(privileges[0])
+
     privileges_dict = modify_privileges(privileges_dict, args)
-    
+
+    error_info = check_tenant_access_set_args(args, privileges_dict)
+    if error_info:
+        return operation_fail(error_info)
+
     error_info = tenant.set_datastore_access_privileges(_auth_mgr.conn, [privileges_dict])
 
     if error_info:
         return operation_fail(error_info)
     else:
         print "tenant access set succeeded"
-       
+
+def check_tenant_access_set_args(args, privileges):
+    """ Check validity of CLI arguments for tenant access set command """
+
+    # If both volume max size and volume total size are NOT set, return directly
+    if (not args.volume_maxsize and not args.volume_totalsize):
+        return None;
+
+    # If both volume max size and volume total size are set,
+    # volume max size should not exceed volume total size
```

Comment:
As you can see in the code logic, check_tenant_access_add_args() is simpler - we are adding a new privilege, so all we need to check is just user input.

The check_tenant_access_set_args(), however, is different and more complicated. We need to check not only user input, but also the existing data in the database.

There's only very minimum duplicate logic in these 2 methods. I feel merging them will cause more confusion than clarity.